### PR TITLE
Remove k8s.io/kubernetes from deps

### PR DESCRIPTION
**Pull request by bryanl**
**What this PR does / why we need it**:

`k8s.io/kubernetes` was not intended to be consumed as a module. This
change removes it and does the following:

* removes the HPA printer due to its reliance on k8s.io/kubernetes. It
can be readded with support for v1 and v2beta2 #930
* updates the kubernetes deps to 1.17.5

**Release note**:
```
Update Kubernetes deps to 1.17.5
```

### CODE REVIEW

File: internal/queryer/queryer.go [hunk 1]
```diff
@@ -848,26 +845,26 @@ func (osq *ObjectStoreQueryer) getSelector(object runtime.Object) (*metav1.Label
...
-	case *apps.ReplicaSet:
-		return t.Spec.Selector, nil
-	case *apps.Deployment:
-		return t.Spec.Selector, nil
-	case *core.Service:
-		selector := &metav1.LabelSelector{
-			MatchLabels: t.Spec.Selector,
-		}
-		return selector, nil
+	// case *apps.DaemonSet:
```
Comment (wwitzel3) [inline internal/queryer/queryer.go:848]: delete

File: vendor/modules.txt [hunk 2]
```diff
@@ -367,7 +378,7 @@ k8s.io/apimachinery/pkg/watch
 k8s.io/apimachinery/third_party/forked/golang/json
 k8s.io/apimachinery/third_party/forked/golang/netutil
 k8s.io/apimachinery/third_party/forked/golang/reflect
-# k8s.io/client-go v0.0.0-20191016230210-14c42cd304d9 => k8s.io/client-go v0.0.0-20190620085101-78d2af792bab
+# k8s.io/client-go v0.17.5
```
Comment (wwitzel3) [inline vendor/modules.txt:381]: 👍 😄

File: go.mod [hunk 3]
```diff
@@ -28,26 +27,22 @@ require (
...
 	google.golang.org/grpc v1.27.0
 	gopkg.in/inf.v0 v0.9.1 // indirect
-	k8s.io/api v0.0.0-20191016225839-816a9b7df678
-	k8s.io/apiextensions-apiserver v0.0.0-20181213153335-0fe22c71c476
-	k8s.io/apimachinery v0.0.0-20191016225534-b1267f8c42b4
-	k8s.io/client-go v0.0.0-20191016230210-14c42cd304d9
+	k8s.io/api v0.17.5
+	k8s.io/apiextensions-apiserver v0.17.5
+	k8s.io/apimachinery v0.17.5
+	k8s.io/client-go v0.17.5
```
Comment (wwitzel3) [inline go.mod:44]: 😄

**Comment by wwitzel3** [comment]
This will need the mocks regenerated.

**Comment by bryanl** [comment]
I'm regenerating these mocks and it's not happy. Will have to dig in.

**Review by wwitzel3 — APPROVED** [review]

**Pull request merged**